lib/ansible/modules/cloud/amazon/ec2_vpc_subnet.py

```diff
@@ -82,152 +89,140 @@
 import time
 
 try:
-    import boto.ec2
-    import boto.vpc
-    from boto.exception import EC2ResponseError
-    HAS_BOTO = True
+    import boto3
+    from botocore.exceptions import ClientError, NoCredentialsError
+    HAS_BOTO3 = True
 except ImportError:
-    HAS_BOTO = False
-    if __name__ != '__main__':
-        raise
-
-from ansible.module_utils.basic import AnsibleModule
-from ansible.module_utils.ec2 import AnsibleAWSError, connect_to_aws, ec2_argument_spec, get_aws_connection_info
-
+    HAS_BOTO3 = False
 
-class AnsibleVPCSubnetException(Exception):
-    pass
 
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import *  # noqa
 
-class AnsibleVPCSubnetCreationException(AnsibleVPCSubnetException):
-    pass
 
+def get_subnet_info(subnet):
+    if 'Subnets' in subnet:
+        return [get_subnet_info(s) for s in subnet['Subnets']]
+    elif 'Subnet' in subnet:
+        subnet = camel_dict_to_snake_dict(subnet['Subnet'])
+    else:
+        subnet = camel_dict_to_snake_dict(subnet)
 
-class AnsibleVPCSubnetDeletionException(AnsibleVPCSubnetException):
-    pass
+    if 'tags' in subnet:
+        subnet['tags'] = boto3_tag_list_to_ansible_dict(subnet['tags'])
+    else:
+        subnet['tags'] = dict()
 
+    if 'subnet_id' in subnet:
```

Comment:
Nice, I was a bit worried the output might change but I see you've thought about that with this.